akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala

```diff
@@ -144,16 +144,23 @@ object BasicPersistentBehaviorCompileOnly {
     }
   // #actor-context
 
+  final case class BookingCompleted(orderNr: String) extends Event
+
   //#snapshottingEveryN
+  import akka.persistence.typed.RetentionCriteria
+
   val snapshottingEveryN = EventSourcedBehavior[Command, Event, State](
     persistenceId = PersistenceId("abc"),
     emptyState = State(),
     commandHandler = (state, cmd) => throw new RuntimeException("TODO: process the command & return an Effect"),
     eventHandler = (state, evt) => throw new RuntimeException("TODO: process the event return the next state"))
-    .snapshotEvery(100)
+    .snapshotWhen {
+      case (_, BookingCompleted(_), _) => true
+      case (_, _, _)                   => false
+    }
+    .withRetention(RetentionCriteria.snapshotEvery(1000))
```

Comment:
Shouldn't the `snapshotWhen` also go into the retention criteria now?
Also - does this really work - will it snapshot both when booking completed and every 1000?